File: java/Preds.java
package irresponsible.crouton;

public class Preds {
  public enum PosInt implements Predicate {
    INSTANCE;
    public final Object test(String s) {
      try {
	if (s.startsWith("-")) return null;
        return new Long(Long.parseLong(s));
      } catch (NumberFormatException e) {
        return null;
      }
    }
  }
  public enum PosHexInt implements Predicate {
    INSTANCE;
    public final Object test(String s) {
      try {
	if (s.startsWith("-")) return null;
        return new Long(Long.parseLong(s, 16));
      } catch (NumberFormatException e) {
        return null;
      }
    }
  }
}

File: java/Preds.kt
package irresponsible.crouton

class Preds {
    enum class PosInt : Predicate {
        INSTANCE;

        override fun test(s: String): Any? {
            try {
                if (s.startsWith("-")) return null
                return java.lang.Long.parseLong(s)
            } catch (e: NumberFormatException) {
                return null
            }

        }
    }

    enum class PosHexInt : Predicate {
        INSTANCE;

        override fun test(s: String): Any? {
            try {
                if (s.startsWith("-")) return null
                return java.lang.Long.parseLong(s, 16)
            } catch (e: NumberFormatException) {
                return null
            }

        }
    }
}
